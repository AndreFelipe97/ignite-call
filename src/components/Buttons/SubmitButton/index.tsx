import { MouseEventHandler, ReactNode } from "react";
import { Button } from "@ignite-ui/react";

interface SubmitButtonProps {
  isDisabled: boolean;
  label: string;
  icon?: ReactNode;
  onClick?: MouseEventHandler<HTMLButtonElement>;
}

export function SubmitButton({
  isDisabled,
  label,
  icon,
  onClick,
}: SubmitButtonProps) {
  return (
    <Button onClick={onClick} type="submit" disabled={isDisabled}>
      {label} {icon}
    </Button>
  );
}
