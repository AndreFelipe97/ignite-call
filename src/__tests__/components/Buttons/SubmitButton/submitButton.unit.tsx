import { render, screen, fireEvent } from "@testing-library/react";
import { SubmitButton } from "@/components/Buttons/SubmitButton";

describe("<SubmitButton />", () => {
  it("should render correctly and active", () => {
    render(<SubmitButton isDisabled={false} label="Test submit button" />);

    const button = screen.getByRole("button");

    expect(button).toBeInTheDocument();
    expect(button).toBeEnabled();
    expect(button).toMatchSnapshot();
  });

  it("should render correctly and with a click function", () => {
    const clickButton = jest.fn();

    render(
      <SubmitButton
        isDisabled={false}
        onClick={clickButton}
        label="Test submit button"
      />
    );

    const button = screen.getByRole("button");

    fireEvent.click(button);

    expect(button).toBeInTheDocument();
    expect(button).toBeEnabled();
    expect(clickButton).toHaveBeenCalledTimes(1);
    expect(button).toMatchSnapshot();
  });

  it("should render correctly and with a icon", () => {
    const iconButton = <span>icon</span>;

    render(
      <SubmitButton
        isDisabled={false}
        label="Test submit button"
        icon={iconButton}
      />
    );

    const icon = screen.getByText("icon");

    expect(icon).toBeInTheDocument();
    expect(icon).toMatchSnapshot();
  });

  it("should render correctly and disable", () => {
    render(<SubmitButton isDisabled={true} label="Test submit button" />);

    const button = screen.getByRole("button");

    expect(button).toBeInTheDocument();
    expect(button).toBeDisabled();
    expect(button).toMatchSnapshot();
  });
});
