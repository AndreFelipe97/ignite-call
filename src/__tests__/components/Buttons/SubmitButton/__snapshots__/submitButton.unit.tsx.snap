// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubmitButton /> should render correctly and active 1`] = `
<button
  class="c-jZXQKc c-jZXQKc-hazApa-variant-primary c-jZXQKc-eDoJcS-size-md"
  type="submit"
>
  Test submit button
   
</button>
`;

exports[`<SubmitButton /> should render correctly and disable 1`] = `
<button
  class="c-jZXQKc c-jZXQKc-hazApa-variant-primary c-jZXQKc-eDoJcS-size-md"
  disabled=""
  type="submit"
>
  Test submit button
   
</button>
`;

exports[`<SubmitButton /> should render correctly and with a click function 1`] = `
<button
  class="c-jZXQKc c-jZXQKc-hazApa-variant-primary c-jZXQKc-eDoJcS-size-md"
  type="submit"
>
  Test submit button
   
</button>
`;

exports[`<SubmitButton /> should render correctly and with a icon 1`] = `
<span>
  icon
</span>
`;
